go-controller/pkg/ovn/egressip.go

```diff
@@ -2451,7 +2451,8 @@ func (e *EgressIPController) deletePodEgressIPAssignment(ni util.NetInfo, egress
 		return err
 	}
 	var ops []ovsdb.Operation
-	if !loadedPodNode || isLocalZonePod { // node is deleted (we can't determine zone so we always try and nuke OR pod is local to zone)
+
+	if !loadedPodNode || isLocalZonePod || (isLocalZoneEgressNode && ni.IsSecondary() && ni.TopologyType() == types.Layer2Topology) { // node is deleted (we can't determine zone so we always try and nuke OR pod is local to zone)
```

Comment:
can you please update commit message to describe what was the stale scenario, why CI was failing etc?

(isLocalZoneEgressNode && ni.IsSecondary() && ni.TopologyType() == types.Layer2Topology)

this bit.. not sure why it needs a specific match here? the 1st two matches are about "not able to determine pod zone" let's try to delete OR "pod is local to this zone" so we must delete the policy...
what's this 3rd case?
I think we should have code comments here for CASE1, CASE2, CASE3 its getting complicated to read

Reply:
sure.

ya, its a mess. 

I was hoping i wasnt the person asked to do it :) Ill do it now.